pkg/okteto/preview.go

```diff
@@ -402,5 +406,25 @@ func (*previewClient) translateErr(err error, name string) error {
 		return oktetoErrors.UserError{E: ErrUnauthorizedGlobalCreation,
 			Hint: "Please log in with an administrator account or use a personal preview environment"}
 	}
+	if err.Error() == "namespace-not-found" {
+		return oktetoErrors.ErrNamespaceNotFound
+	}
 	return err
 }
+
+// Get gets the given preview environment, returns its ID if found
+func (c *previewClient) Get(ctx context.Context, previewName string) (*types.Preview, error) {
+	queryStruct := getPreviewQuery{}
+
+	variables := map[string]interface{}{
+		"id": graphql.String(previewName),
+	}
+	err := query(ctx, &queryStruct, variables, c.client)
+	if err != nil {
+		return nil, err
```

Comment:
I think you need to call here to `translateErr`, if not, the change you did in that function won't be applied in this call, where I believe you need it

Reply:
`translateErr` is called prior exiting query or mutation functions, so the error is transformed before.

I was investigating this and the cause was the error thrown by the API when a namespace is a preview and we query `space` is "not-found" which does not fall into `namespace-not-found` so it was not looking for the preview.

I added some changes and tests to check this.

I've been doing some smoke test and i fix it. At the capture, the namespace creating gives the following error at API

`ERRO[3566] failed to create space teresaromero-caracolaaa  error="failed to create rolebindings for teresaromero-caracolaaa: error creating kubernetes role binding: RoleBinding.rbac.authorization.k8s.io \"teresaromero-caracolaaa\" is invalid: roleRef.name: Required value" http.request.id=e9f2ba6c-7ee9-4485-af8e-8e3aed188c05 instance.id=teresaromero-okteto-api-okteto-6946d9fcf6-9dqh2 user.id=okteto-admin`

Let me know if you find something else. Thanks!
![Screenshot 2024-06-21 at 08 43 26](https://github.com/okteto/okteto/assets/40767058/cfc08710-e1c6-4b7a-a94d-a02337491c2a)

Reply:
The error about the namespace creation should not happen. Was it that in your dev env? Was it updated with master?

Reply:
i re-tested, it was puntual error from my dev